rgw/v2/tests/s3_swift/test_list_all_multipart_uploads.py

```diff
@@ -91,102 +100,148 @@ def test_exec(config, ssh_con):
+                bucket_name, rgw_tenant1_user1, tenant1_user1_info
+            )
+            if config.test_ops.get("enable_version", False):
+                log.info(f"bucket versionig test on bucket: {bucket.name}")
+                reusable.enable_versioning(
+                    bucket, rgw_tenant1_user1, tenant1_user1_info, write_bucket_io_info
                 )
-        srv_time_post_op = get_svc_time(ssh_con)
-        log.info(srv_time_pre_op)
-        log.info(srv_time_post_op)
-
-        if srv_time_post_op > srv_time_pre_op:
-            log.info("Service is running without crash")
-        else:
-            raise TestExecError("Service got crashed")
-
-    # get policy
-    get_policy = rgw_tenant1_user1_c.get_bucket_policy(Bucket=t1_u1_bucket1.name)
-    log.info("got bucket policy:%s\n" % get_policy["Policy"])
-
-    # List multipart uploads with tenant1_user2 user with bucket t1_u1_bucket1
-    multipart_object1 = rgw_tenant1_user2_c.list_multipart_uploads(
-        Bucket=t1_u1_bucket1.name
-    )
-    log.info("Multipart object %s" % multipart_object1)
 
-    # Verify tenant1_user2 not having permission for listing multipart uploads in t1_u1_bucket2
-    try:
-        multipart_object2 = rgw_tenant1_user2_c.list_multipart_uploads(
-            Bucket=t1_u1_bucket2.name
-        )
-        raise Exception(
-            "%s user should not list multipart uploads in bucket: %s"
-            % (tenant1_user2_info["user_id"], t1_u1_bucket2.name)
-        )
-    except ClientError as err:
-        log.error("Listing failed as expected with exception: %s" % err)
+            bucket_policy_generated = s3_bucket_policy.gen_bucket_policy(
+                tenants_list=[tenant1],
+                userids_list=[tenant1_user2_info["user_id"]],
+                actions_list=action_list,
+                resources=[bucket_name],
+            )
+
+            if config.test_ops.get("list_bucket_versions", False):
+                bucket_policy_generated["Statement"][0]["Principal"][
+                    "AWS"
+                ] += user3_aws_principle
+
+            bucket_policy = json.dumps(bucket_policy_generated)
+            log.info(f"bucket_policy_generated :{bucket_policy}")
+            bucket_policy_obj = s3lib.resource_op(
+                {
+                    "obj": rgw_tenant1_user1,
+                    "resource": "BucketPolicy",
+                    "args": [bucket_name],
+                }
+            )
+            put_policy = s3lib.resource_op(
+                {
+                    "obj": bucket_policy_obj,
+                    "resource": "put",
+                    "kwargs": dict(
+                        ConfirmRemoveSelfBucketAccess=True, Policy=bucket_policy
+                    ),
+                }
+            )
+            log.info(f"put policy response: {put_policy}\n")
+            if put_policy is False:
+                raise TestExecError("Resource execution failed: bucket creation faield")
```

Comment:
Can we have a more reliable error message

Reply:
done